R/tfrmt_checks.R

```diff
@@ -284,3 +284,24 @@ check_footnote_plan <- function(x){
 
 }
 
+
+#' Check if the plan parameter is consistent in tfrmt
+#' @noRd
+#' @param tfrmt_object tfrmt object to be checked
+#' @param plan plan parameter, e.g., col_style_plan
+#' @importFrom rlang abort inherits
+#'
+#'
+check_plan <- function(tfrmt_object, plan) {
+  # extract the plan element from the tfrmt_object
+  plan_element <- tfrmt_object[[plan]]
+
+  # check if the user supplied a value to the plan parameter
+  if (!is.null(plan_element)) {
+    # Check if the plan element inherits the plan attribute
+    if (!inherits(plan_element, plan)) {
+      stop("Check if ", plan, " parameter has been supplied ", plan, "() function")
```

Comment:
Can you please revise to be a bit clearer about what is wrong, e.g. that an invalid *_plan is supplied to the argument

Reply:
Does the new error message in (https://github.com/GSK-Biostatistics/tfrmt/pull/428/commits/b9abc09dd3ca6a937e6cc504588eab8c70fbcd5e) sound better?

![image](https://github.com/GSK-Biostatistics/tfrmt/assets/72526731/6d8a912a-6929-4a37-bb9e-2fcac380f979)

Reply:
I think just "Invalid input supplied to the `col_style_plan` parameter. Please supply a `col_style_plan()`.